### ISSUE TITLE
WARNING: Could not parse *** function

### ISSUE THREAD

**Issue by skick1234** [distubejs/ytdl-core#144]
### Don't create a new issue for this warning.

**Comment by brianbuie** [comment]
Tried updating from 4.15.1 to 4.15.3 on Node 22.11, but this error keeps dumping 1000's of lines of debug info in the console after every request.

Everything seems to be working fine, is there a way to suppress this warning?

**Comment by Hoshimikan6490** [comment]
Please check change of #147 . It show us how to fix it.

In addition, you should not discuss about the issue in here because this issue is for "WARNING: Could not parse *** function". If admins see this comments, I think they should remove or hide these comments(include this comment).

**Comment by ziluotian** [comment]
The data that are successfully parsed in the current code are all returned by API requests. Parsing the data in html5player without using the API fails. Please fix the sig.decipherFormats function so that it can parse the data in html5player.

**Comment by ziluotian** [comment]
I got the *** function, but the parsed link returns 403. 
![截屏2024-12-17 14 46 32](https://github.com/user-attachments/assets/6079dd49-6bcc-4e89-a15f-0bb8bc387eca)

**Comment by ziluotian** [comment]
Only the link from itag18 can be played, other links return 403, is there any way to decrypt other links?

**Comment by v99d** [comment]
i got this error 

```
WARNING: Could not parse n transform function.
Please report this issue with the "1735137353847-base.js" file on https://github.com/distubejs/ytdl-core/issues.
Error [PlayingError]: FFmpeg/avconv not found!
Id: 5uh9imWsNNY
Name: Nour & @Eldab3 - Meen Ysadak | Official Audio - 2024 | نور و الضبع - مين يصدق
    at FFmpeg.getInfo (C:\Users\Hamza\Desktop\music updated\node_modules\distube\node_modules\prism-media\src\core\FFmpeg.js:143:11)
    at FFmpeg.create (C:\Users\Hamza\Desktop\music updated\node_modules\distube\node_modules\prism-media\src\core\FFmpeg.js:156:38)
    at new FFmpeg (C:\Users\Hamza\Desktop\music updated\node_modules\distube\node_modules\prism-media\src\core\FFmpeg.js:45:27)
    at new DisTubeStream (C:\Users\Hamza\Desktop\music updated\node_modules\distube\dist\core\DisTubeStream.js:70:23)
    at DisTubeStream.YouTube (C:\Users\Hamza\Desktop\music updated\node_modules\distube\dist\core\DisTubeStream.js:88:16)
    at DisTubeHandler.createStream (C:\Users\Hamza\Desktop\music updated\node_modules\distube\dist\core\DisTubeHandler.js:335:37)
    at QueueManager.playSong (C:\Users\Hamza\Desktop\music updated\node_modules\distube\dist\core\manager\QueueManager.js:83:41)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async QueueManager.create (C:\Users\Hamza\Desktop\music updated\node_modules\distube\dist\core\manager\QueueManager.js:45:25)
    at async DisTube.play (C:\Users\Hamza\Desktop\music updated\node_modules\distube\dist\DisTube.js:197:42)
    ```

**Comment by Hoshimikan6490** [comment]
> FFmpeg/avconv not found!

Did you installed FFmpeg?
https://github.com/skick1234/DisTube/wiki/DisTube-Guide#installation

**Comment by RM7Shmr** [comment]
I have a problem when I write a command to play the song. Sometimes it sends that the song has been played, but the bot does not play the song, and sometimes it does play it. Why? 
I use distube library

**Comment by Pizzy23** [comment]
Plls help

```
[2025/01/24 08:56:51] WARNING: Could not parse n transform function.
Please report this issue with the "1737719811710-base.js" file on https://github.com/distubejs/ytdl-core/issues.
[2025/01/24 08:56:52] Queue {
  distube: <ref *1> DisTube {
    _events: [Object: null prototype] {
      playSong: [Function (anonymous)],
      addSong: [Function (anonymous)],
      addList: [Function (anonymous)],
      error: [Function (anonymous)],
      empty: [Function (anonymous)],
      searchNoResult: [Function (anonymous)],
      finish: [Function (anonymous)]
    },
    _eventsCount: 7,
    _maxListeners: 15,
    handler: DisTubeHandler { distube: [Circular *1] },
    options: Options {
      plugins: [Array],
      emitNewSongOnly: false,
      savePreviousSongs: true,
      customFilters: undefined,
      nsfw: true,
      emitAddSongWhenCreatingQueue: false,
      emitAddListWhenCreatingQueue: false,
      joinNewVoiceChannel: true,
      ffmpeg: [Object]
    },
    client: Client {
      _events: [Object: null prototype],
      _eventsCount: 5,
      _maxListeners: undefined,
      options: [Object],
      rest: [REST],
      ws: [WebSocketManager],
      actions: [ActionsManager],
      voice: [ClientVoiceManager],
      shard: null,
      users: UserManager {},
      guilds: GuildManager {},
      channels: ChannelManager {},
      sweepers: [Sweepers],
      presence: [ClientPresence],
      user: [ClientUser],
      application: [ClientApplication],
      readyTimestamp: 1737719801501,
      important: [Object],
      embed: [Object],
      distube: [Circular *1],
      config: [Object],
      commands: [Collection [Map]],
      status: [Object],
      [Symbol(shapeMode)]: false,
      [Symbol(kCapture)]: true
    },
    queues: QueueManager {
      distube: [Circular *1],
      collection: [Collection [Map]]
    },
    voices: DisTubeVoiceManager {
      distube: [Circular *1],
      collection: [Collection [Map]]
    },
    plugins: [
      [SpotifyPlugin],
      [YouTubePlugin],
      [SpotifyPlugin],
      [SoundCloudPlugin]
    ],
    filters: {
      '3d': 'apulsator=hz=0.125',
      bassboost: 'bass=g=10',
      echo: 'aecho=0.8:0.9:1000:0.3',
      flanger: 'flanger',
      gate: 'agate',
      haas: 'haas',
      karaoke: 'stereotools=mlev=0.1',
      nightcore: 'asetrate=48000*1.25,aresample=48000,bass=g=5',
      reverse: 'areverse',
      vaporwave: 'asetrate=48000*0.8,aresample=48000,atempo=1.1',
      mcompand: 'mcompand',
      phaser: 'aphaser',
      tremolo: 'tremolo',
      surround: 'surround',
      earwax: 'earwax'
    },
    [Symbol(shapeMode)]: false,
    [Symbol(kCapture)]: false
  },
  id: '558306899288915970',
  voice: DisTubeVoice {
    _events: [Object: null prototype] {
      disconnect: [Function: disconnect],
      error: [Function: error],
      finish: [Function: finish]
    },
    _eventsCount: 3,
    _maxListeners: undefined,
    id: '558306899288915970',
    voices: DisTubeVoiceManager {
      distube: [DisTube],
      collection: [Collection [Map]]
    },
    audioPlayer: AudioPlayer {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      _state: [Object],
      subscribers: [Array],
      behaviors: [Object],
      debug: [Function (anonymous)],
      [Symbol(shapeMode)]: false,
      [Symbol(kCapture)]: false
    },
    connection: VoiceConnection {
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      rejoinAttempts: 0,
      _state: [Object],
      joinConfig: [Object],
      packets: [Object],
      receiver: [VoiceReceiver],
      debug: null,
      onNetworkingClose: [Function: bound onNetworkingClose],
      onNetworkingStateChange: [Function: bound onNetworkingStateChange],
      onNetworkingError: [Function: bound onNetworkingError],
      onNetworkingDebug: [Function: bound onNetworkingDebug],
      [Symbol(shapeMode)]: false,
      [Symbol(kCapture)]: false
    },
    emittedError: true,
    isDisconnected: false,
    stream: DisTubeStream {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      process: [ChildProcess],
      stream: [VolumeTransformer],
      audioResource: [AudioResource],
      [Symbol(shapeMode)]: false,
      [Symbol(kCapture)]: false
    },
    [Symbol(shapeMode)]: false,
    [Symbol(kCapture)]: false
  },
  songs: [],
  previousSongs: [],
  stopped: false,
  playing: true,
  paused: false,
  repeatMode: 0,
  autoplay: false,
  ffmpegArgs: { global: {}, input: {}, output: {} },
  textChannel: <ref *2> TextChannel {
    type: 0,
    guild: Guild {
      id: '558306899288915970',
      name: 'RPG',
      icon: '8df51be7e9ecc942922236206750332c',
      features: [Array],
      commands: [GuildApplicationCommandManager],
      members: [GuildMemberManager],
      channels: [GuildChannelManager],
      bans: [GuildBanManager],
      roles: [RoleManager],
      presences: PresenceManager {},
      voiceStates: [VoiceStateManager],
      stageInstances: [StageInstanceManager],
      invites: [GuildInviteManager],
      scheduledEvents: [GuildScheduledEventManager],
      autoModerationRules: [AutoModerationRuleManager],
      available: true,
      shardId: 0,
      splash: null,
      banner: null,
      description: null,
      verificationLevel: 0,
      vanityURLCode: null,
      nsfwLevel: 0,
      premiumSubscriptionCount: 1,
      discoverySplash: null,
      memberCount: 18,
      large: false,
      premiumProgressBarEnabled: false,
      applicationId: null,
      afkTimeout: 300,
      afkChannelId: null,
      systemChannelId: null,
      premiumTier: 0,
      widgetEnabled: null,
      widgetChannelId: null,
      explicitContentFilter: 0,
      mfaLevel: 0,
      joinedTimestamp: 1635810235303,
      defaultMessageNotifications: 0,
      systemChannelFlags: [SystemChannelFlagsBitField],
      maximumMembers: 500000,
      maximumPresences: null,
      maxVideoChannelUsers: 25,
      maxStageVideoChannelUsers: 50,
      approximateMemberCount: null,
      approximatePresenceCount: null,
      vanityURLUses: null,
      rulesChannelId: null,
      publicUpdatesChannelId: null,
      preferredLocale: 'en-US',
      safetyAlertsChannelId: null,
      ownerId: '229724269150470144',
      emojis: [GuildEmojiManager],
      stickers: [GuildStickerManager]
    },
    guildId: '558306899288915970',
    permissionOverwrites: PermissionOverwriteManager { channel: [Circular *2] },
    messages: GuildMessageManager { channel: [Circular *2] },
    threads: GuildTextThreadManager { channel: [Circular *2] },
    nsfw: false,
    flags: ChannelFlagsBitField { bitfield: 0 },
    id: '1027763151845990431',
    name: 'test',
    rawPosition: 20,
    parentId: '741860512316522546',
    topic: null,
    lastMessageId: '1332318199844175933',
    defaultThreadRateLimitPerUser: null,
    rateLimitPerUser: 0
  },
  _beginTime: 0,
  _next: false,
  _prev: false,
  _taskQueue: TaskQueue {},
  _listeners: {
    disconnect: [Function: disconnect],
    error: [Function: error],
    finish: [Function: finish]
  }
}
```

**Comment by Cocaina-Ricky** [comment]
hey. just tried 14.6.3, still getting same problem. All packages are installed, everything connects fine, but once a song is played, that error is given.

**Comment by ryokr** [comment]
WARNING: Could not parse n transform function. @skick1234 

[1740915773459-base.zip](https://github.com/user-attachments/files/19042826/1740915773459-base.zip)

**Comment by Keito-Klein** [comment]
**Error**
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1740958507628-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

**Version**
"@distube/ytdl-core": "^4.16.4"

**Code**
```js
const agent = ytdl.createAgent(cookies);

if (format == "mp3") {
    return new Promise((resolve, reject) => {
      ytdl.getInfo(url, { agent }).then((api) => {
        const array = [];
        for (let i = 0; i < api.formats.length; i++) {
          if (
            api.formats[i].codecs === "opus" &&
            api.formats[i].audioQuality === "AUDIO_QUALITY_MEDIUM"
          ) {
            array.push({
              title: api.videoDetails.title,
              channel: api.videoDetails.author.name,
              subscriber: api.videoDetails.author.subscriber_count,
              viewCount: api.videoDetails.viewCount,
              likes: api.videoDetails.likes,
              audioBitrate: api.formats[i].audioBitrate,
              url: api.formats[i].url,
            });
          }
        }
        if (array.length === 0) reject("no audio available");
        const highesBitrate = array.reduce((current, highes) => {
          return current.audioBitrate > highes.audioBitrate ? current : highes;
        }, array[0]);

        resolve(highesBitrate);
      });
    });
  }
```

[1740958507628-base.zip](https://github.com/user-attachments/files/19045297/1740958507628-base.zip)

**Comment by Quema100** [comment]
I deleted the file "1741063839158-base.js" and ran the bot again, and the issue disappeared.

```ps1 
WARNING: Could not parse n transform function.
  Please report this issue by uploading the "1741063839158-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
  0 0 0
  https://youtu.be/gyuLluLtRr0?si=0f_Jvs6jkle0oqBZ
  User settings: ko
  error occurred: evalmachine.<anonymous>:3
  Tu:function(k){k.reverse()}};var ik9=function(k){k=k.split(tw[2]);cP.K5(k,2);cP.h2(k,53);cP.K5(k,3);cP.Tu(k,80);cP.K5(k,2);cP.h2(k,21);return k.join(tw[2])};ik9(sig);
                                                             ^
  
  ReferenceError: tw is not defined
      at ik9 (evalmachine.<anonymous>:3:60)
      at evalmachine.<anonymous>:3:158
      at Script.runInContext (node:vm:149:12)
      at Script.runInNewContext (node:vm:154:17)
      at decipher (C:\Users\myoon\Documents\STAFF-Bot\node_modules\@distube\ytdl-core\lib\sig.js:235:66)
      at exports.setDownloadURL (C:\Users\myoon\Documents\STAFF-Bot\node_modules\@distube\ytdl-core\lib\sig.js:249:36)
      at C:\Users\myoon\Documents\STAFF-Bot\node_modules\@distube\ytdl-core\lib\sig.js:265:13
      at Array.forEach (<anonymous>)
      at exports.decipherFormats (C:\Users\myoon\Documents\STAFF-Bot\node_modules\@distube\ytdl-core\lib\sig.js:264:11)
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
```

Is it okay to delete it?

**Comment by bkmkmon** [comment]
[1742479982647-base.zip](https://github.com/user-attachments/files/19369681/1742479982647-base.zip)
Could not parse n transform function.

**Comment by xMiyuka** [comment]
[1742530393150-base.js.zip](https://github.com/user-attachments/files/19381040/1742530393150-base.js.zip)

**Comment by artem-pereverzev** [comment]
[1742552407123-base.js.zip](https://github.com/user-attachments/files/19387325/1742552407123-base.js.zip)

**Comment by bkmkmon** [comment]
Could not parse n transform function.
4.16.5

[1742837200919-base.zip](https://github.com/user-attachments/files/19434836/1742837200919-base.zip)

**Comment by plasma119** [comment]
Could not parse n transform function.
4.16.5

[1742839463707-base.zip](https://github.com/user-attachments/files/19435634/1742839463707-base.zip)
video id: 6r0D_XSaSFs

evalmachine.<anonymous>:3
sd:function(p){p.reverse()}};var aZP=function(p){p=p.split(lP[4]);ky.Eh(p,29);ky.Rr(p,3);ky.Eh(p,47);ky.Rr(p,3);ky.Eh(p,24);ky.Eh(p,54);ky.sd(p,65);return p.join(lP[4])};aZP(sig);
                                                           ^

ReferenceError: lP is not defined
    at aZP (evalmachine.<anonymous>:3:60)
    at evalmachine.<anonymous>:3:171
    at Script.runInContext (node:vm:149:12)
    at Script.runInNewContext (node:vm:154:17)
    at decipher (D:\Youtube Archive\node_modules\@distube\ytdl-core\lib\sig.js:244:66)
    at exports.setDownloadURL (D:\Youtube Archive\node_modules\@distube\ytdl-core\lib\sig.js:258:36)
    at D:\Youtube Archive\node_modules\@distube\ytdl-core\lib\sig.js:274:13
    at Array.forEach (<anonymous>)
    at exports.decipherFormats (D:\Youtube Archive\node_modules\@distube\ytdl-core\lib\sig.js:273:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)

**Comment by migferamiral** [comment]
WARNING: Could not parse n transform function.
Version: 4.16.5

[1742840361087-base.zip](https://github.com/user-attachments/files/19435855/1742840361087-base.zip)

**Comment by jiejie39** [comment]
WARNING: Could not parse n transform function.
Version: 4.16.5

[1742842848126-base.zip](https://github.com/user-attachments/files/19436549/1742842848126-base.zip)

**Comment by actuallyatiger** [comment]
WARNING: Could not parse n transform function.
Version: 4.16.5

[1742844274351-base.zip](https://github.com/user-attachments/files/19436808/1742844274351-base.zip)

**Comment by Dorian399** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1742844160354-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
FFmpeg error: Error: Input stream error: Status code: 403

[1742844160354-base.zip](https://github.com/user-attachments/files/19436811/1742844160354-base.zip)

**Comment by ab42jj** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1742848511592-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

**Comment by hpq-dev** [comment]
[1742849296942-base.zip](https://github.com/user-attachments/files/19438003/1742849296942-base.zip)

**Comment by Noahcatz** [comment]
[1742852615978-base.zip](https://github.com/user-attachments/files/19438531/1742852615978-base.zip)

**Comment by Yongrui-Su** [comment]
[1742855143923-base.zip](https://github.com/user-attachments/files/19438854/1742855143923-base.zip)

**Comment by victorsouzaleal** [comment]
[1742856904352-base.zip](https://github.com/user-attachments/files/19439101/1742856904352-base.zip)

**Comment by navarog3** [comment]
[1742856625247-base.zip](https://github.com/user-attachments/files/19439110/1742856625247-base.zip)

**Comment by Techers** [comment]
[1742864892156-base.zip](https://github.com/user-attachments/files/19440084/1742864892156-base.zip)
WARNING: Could not parse n transform function.

**Comment by arty01238** [comment]
I now have this issue also, everything was fine until today.

**Comment by nguyenluu1998** [comment]
@skick1234 Hi, is there any update on this ?

**Comment by bnfone** [comment]
> I now have this issue also, everything was fine until today.

same here!

**Comment by ProtectionPlus-Owner** [comment]
@skick1234 

[1742915688143-base.txt](https://github.com/user-attachments/files/19451187/1742915688143-base.txt)

![Image](https://github.com/user-attachments/assets/a9021845-aa01-4011-aac3-d2417cb7fb56)

**Comment by luxferooo** [comment]
same theme 

[1742918764242-base.txt](https://github.com/user-attachments/files/19451796/1742918764242-base.txt)

**Comment by CaquinhoDev** [comment]
WARNING: Could not parse n transform function.

[1742920391582-base.js.txt](https://github.com/user-attachments/files/19452242/1742920391582-base.js.txt)

**Comment by BySkyles** [comment]
[1742942846481-base.txt](https://github.com/user-attachments/files/19457856/1742942846481-base.txt)

**Comment by kevinmnm** [comment]
[1742948122388-base.txt](https://github.com/user-attachments/files/19458547/1742948122388-base.txt)

**Comment by Tati-Mendonca** [comment]
[1742965232248-base.txt](https://github.com/user-attachments/files/19460709/1742965232248-base.txt)

**Comment by Yamac-Koc** [comment]
(node:3216) DeprecationWarning: Guild#me is deprecated. Use Guild#members#me instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
WARNING: @distube/ytsr is out of date! Update with "npm install @distube/ytsr@latest".
Found 1 successful responses from clients
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1742968533053-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1742968533053-base.txt](https://github.com/user-attachments/files/19461292/1742968533053-base.txt)

it was fine please fix it!!

**Comment by PositiveRocket** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1742971103402-base.js" file on [#144](https://github.com/distubejs/ytdl-core/issues/144).

[1742970529577-base.zip](https://github.com/user-attachments/files/19461721/1742970529577-base.zip)

**Comment by hyongsu87** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1742979403673-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1737367811715.log](https://github.com/user-attachments/files/19463551/1737367811715.log)
Uncaught UnhandledPromiseRejection UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "#<WriteStream>".
    at throwUnhandledRejectionsMode (<node_internals>/internal/process/promises:392:7)
    at processPromiseRejections (<node_internals>/internal/process/promises:475:17)
    at processTicksAndRejections (<node_internals>/internal/process/task_queues:106:32)
promises:392
Process exited with code 1

Hey, thanks so much for everything!

**Comment by hyongsu87** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1742979403673-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
Uncaught UnhandledPromiseRejection UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "#<WriteStream>".
    at throwUnhandledRejectionsMode (<node_internals>/internal/process/promises:392:7)
    at processPromiseRejections (<node_internals>/internal/process/promises:475:17)
    at processTicksAndRejections (<node_internals>/internal/process/task_queues:106:32)
promises:392
Process exited with code 1

[1742979403673-base.zip](https://github.com/user-attachments/files/19463568/1742979403673-base.zip)

Thanks so much for what you're doing!

**Comment by AlfiCloud** [comment]
[1743002558191-base.zip](https://github.com/user-attachments/files/19470653/1743002558191-base.zip)
why this?

**Comment by Daniyal-Ali1043** [comment]
[1743010832914-base.js.tar.gz](https://github.com/user-attachments/files/19473052/1743010832914-base.js.tar.gz)

**Comment by HolyZephyros** [comment]
[1743019796407-base.zip](https://github.com/user-attachments/files/19474949/1743019796407-base.zip)

same here.

**Comment by SubArrest** [comment]
[1743026208843-base.zip](https://github.com/user-attachments/files/19476006/1743026208843-base.zip)

**Comment by luanargolodev** [comment]
[1743030965011-base.zip](https://github.com/user-attachments/files/19476545/1743030965011-base.zip)

I'm having this problem

**Comment by pashalecqua** [comment]
WARNING: Could not parse n transform function.

[1743031319300-base.zip](https://github.com/user-attachments/files/19477047/1743031319300-base.zip)

**Comment by gathilas** [comment]
WARNING: Could not parse n transform function.

[1743082037151-base.zip](https://github.com/user-attachments/files/19487177/1743082037151-base.zip)

**Comment by amir18160** [comment]
[1743084537757-base.zip](https://github.com/user-attachments/files/19488380/1743084537757-base.zip)

its not working at all anymore.

**Comment by cjmartin** [comment]
/Users/cjmartin/Code/yt-fetcher/1743091999224-base.js

**Comment by cjmartin** [comment]
Also

WARNING: Could not parse n transform function.

[1743091999224-base.js.zip](https://github.com/user-attachments/files/19490108/1743091999224-base.js.zip)

**Comment by niyahall** [comment]
WARNING: Could not parse n transform function

[1743125812385-base.zip](https://github.com/user-attachments/files/19496827/1743125812385-base.zip)

**Comment by Julzies** [comment]
Could not parse n transform function

[1743129416573-base.zip](https://github.com/user-attachments/files/19497392/1743129416573-base.zip)

**Comment by jjeantw** [comment]
WARNING: Could not parse n transform function
[1743157157225-base.js.zip](https://github.com/user-attachments/files/19503965/1743157157225-base.js.zip)

**Comment by Sakshi-Khutwad** [comment]
I am using distube/ytdl-core and the fowllowing error is encountered

WARNING: Could not parse n transform function.
Please report this issue by uploading the "1743178195538-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

**Comment by BobbyHertz** [comment]
+1

**Comment by buoy1897** [comment]
Using it for Muse Bot and got the same error as everyone else. Thanks for your work!

[1743193156788-base.zip](https://github.com/user-attachments/files/19511946/1743193156788-base.zip)

**Comment by dhlah** [comment]
Found 2 successful responses from clients
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1743204406671-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1743204406671-base.zip](https://github.com/user-attachments/files/19513922/1743204406671-base.zip)

**Comment by Pizzy23** [comment]
[1743207917863-base.zip](https://github.com/user-attachments/files/19514230/1743207917863-base.zip)

[2025/03/28 21:25:17] WARNING: Could not parse n transform function.
Please report this issue with the "1743207917863-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

**Comment by kohbo** [comment]
[1743215352144-base.tar.gz](https://github.com/user-attachments/files/19514846/1743215352144-base.tar.gz)

**Comment by SerCrAsH** [comment]
Please check this @BobbyHertz solution:
https://github.com/distubejs/ytdl-core/issues/213#issuecomment-2763289092

**Comment by HolyZephyros** [comment]
> Please check this [@BobbyHertz](https://github.com/BobbyHertz) solution: [#213 (comment)](https://github.com/distubejs/ytdl-core/issues/213#issuecomment-2763289092)

It worked for the moment until the main patch comes. Thanks.

**Comment by osistlk** [comment]
```
Found 2 successful responses from clients
WARNING: Could not parse n transform function.
```

[1743253515934-base.js.gz](https://github.com/user-attachments/files/19521235/1743253515934-base.js.gz)

**Comment by warior456** [comment]
[1743258887594-base.js.gz](https://github.com/user-attachments/files/19521598/1743258887594-base.js.gz)

**Comment by vijayjvvc** [comment]
> ### Please upload your debug file in the comment.
> ### Don't create

[1743311003700-base.js.gz](https://github.com/user-attachments/files/19524319/1743311003700-base.js.gz)

 a new issue for this warning.

**Comment by monir6163** [comment]
every request this file generated how to fix it 1743319835563-base.js

**Comment by Quema100** [comment]
umm..
WARNING: Could not parse decipher function.
Stream URLs will be missing.
Please report this issue with the "1743336681087-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
WARNING: Could not parse n transform function.
Please report this issue with the "1743336683343-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1743336681087-base.zip](https://github.com/user-attachments/files/19525569/1743336681087-base.zip)

**Comment by technic05** [comment]
```
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1743357040705-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
```

[1743357040705-base.zip](https://github.com/user-attachments/files/19526890/1743357040705-base.zip)

**Comment by equizno345** [comment]
Found 2 successful responses from clients
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1743362403670-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
Found 2 successful responses from clients

[1743362403670-base.zip](https://github.com/user-attachments/files/19527216/1743362403670-base.zip)

**Comment by Mvb1122** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1743376838031-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1743376838031-base.zip](https://github.com/user-attachments/files/19527926/1743376838031-base.zip)

**Comment by dev-alto** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1743395153365-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
ReferenceError: tj is not defined

[1743395153365-base.zip](https://github.com/user-attachments/files/19529666/1743395153365-base.zip)

**Comment by shm4r7** [comment]
Here you are
[1743528125958-base.zip](https://github.com/user-attachments/files/19555265/1743528125958-base.zip)

**Comment by Ruuroka** [comment]
[1744991192253-base.zip](https://github.com/user-attachments/files/19817003/1744991192253-base.zip)

Can you please do something with this? I updated the ytdl and still cannot download

**Comment by SellingVika7777** [comment]
Before everything worked

0|bot      | WARNING: Could not parse decipher function.
0|bot      | Stream URLs will be missing.
0|bot      | Please report this issue by uploading the "1745342492968-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
0|bot      | WARNING: Could not parse n transform function.
0|bot      | Please report this issue by uploading the "1745342492979-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
0|bot      | DisTubeError [FFMPEG_EXITED]: ffmpeg exited with code 1
0|bot      |     at ChildProcess.<anonymous> (/root/nyx/node_modules/distube/dist/index.js:904:26)
0|bot      |     at ChildProcess.emit (node:events:518:28)
0|bot      |     at ChildProcess._handle.onexit (node:internal/child_process:293:12) {
0|bot      |   errorCode: 'FFMPEG_EXITED'
0|bot      | }
[files.zip](https://github.com/user-attachments/files/19854343/files.zip)

**Comment by Dorian399** [comment]
[33mWARNING:[0m Could not parse decipher function.
Stream URLs will be missing.
Please report this issue by uploading the "1745347811732-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
[33mWARNING:[0m Could not parse n transform function.
Please report this issue by uploading the "1745347811740-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
FFmpeg error: Error: Input stream error: Status code: 403
    at PassThrough.<anonymous> (redacted/yt-api/node_modules/fluent-ffmpeg/lib/processor.js:451:34)
    at PassThrough.emit (node:events:531:35)
    at PassThrough.emit (node:events:519:28)
    at ClientRequest.<anonymous> (redacted/yt-api/node_modules/miniget/dist/index.js:211:28)
    at Object.onceWrapper (node:events:634:26)
    at ClientRequest.emit (node:events:519:28)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:709:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
    at TLSSocket.socketOnData (node:_http_client:551:22)
    at TLSSocket.emit (node:events:519:28) {
  inputStreamError: MinigetError: Status code: 403
      at ClientRequest.<anonymous> (redacted/yt-api/node_modules/miniget/dist/index.js:206:27)
      at Object.onceWrapper (node:events:634:26)
      at ClientRequest.emit (node:events:519:28)
      at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:709:27)
      at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
      at TLSSocket.socketOnData (node:_http_client:551:22)
      at TLSSocket.emit (node:events:519:28)
      at addChunk (node:internal/streams/readable:559:12)
      at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
      at Readable.push (node:internal/streams/readable:390:5) {
    statusCode: 403

[debug.zip](https://github.com/user-attachments/files/19855494/debug.zip)

**Comment by HolyZephyros** [comment]
WARNING: Could not parse decipher function.
 Stream URLs will be missing.
 Please report this issue by uploading the "1745353697259-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1745353697270-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[Desktop.zip](https://github.com/user-attachments/files/19856432/Desktop.zip)

**Comment by RafaelSantos777** [comment]
WARNING: Could not parse decipher function.
Stream URLs will be missing.
Please report this issue by uploading the "1745362992473-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1745362992477-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1745362992477-player-script.zip](https://github.com/user-attachments/files/19857956/1745362992477-player-script.zip)

**Comment by BobbyHertz** [comment]
Same, broke again earlier this afternoon.

**Comment by dev25-hs** [comment]
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1745390373096-player-script.js" file on https://github.com/distubejs/ytdl

[1745390053008-player-script.txt](https://github.com/user-attachments/files/19862270/1745390053008-player-script.txt)

-core/issues/144.

**Comment by vkq2808** [comment]
[C:\Users\khanhquoc\Desktop\project\graduation-project\be\1745396614726-player-script.js](url)

WARNING: Could not parse n transform function.
Please report this issue by uploading the "1745396614726-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

**Comment by LordCat0** [comment]
This is never going to be fixed is it

**Comment by LordCat0** [comment]
WARNING: Could not parse decipher function.
Stream URLs will be missing.
Please report this issue by uploading the "1745436100139-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1745436100139-player-script.zip](https://github.com/user-attachments/files/19874716/1745436100139-player-script.zip)

WARNING: Could not parse n transform function.
Please report this issue by uploading the "1745436100145-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1745436100145-player-script.zip](https://github.com/user-attachments/files/19874721/1745436100145-player-script.zip)

**Comment by BobbyHertz** [comment]
It's absolutely ridiculous how unstable this solution is.

It shouldn't be broken like this every other week due to minor YouTube changes.

**Comment by HolyZephyros** [comment]
If anyone has a temporary solution, it would be great if they could share it.

**Comment by Quema100** [comment]
I think we need to find another library to replace this one.

**Comment by Method-Development** [comment]
Anyone found a work around? Bot is crashing out still :(

**Comment by David-237-9** [comment]
> ### Please upload your debug file in the comment.
> ### Don't create a new issue for this warning.

[1745524555403-player-script.zip](https://github.com/user-attachments/files/19898298/1745524555403-player-script.zip)

**Comment by David-237-9** [comment]
2 problems here:

[1745524555403-player-script.zip](https://github.com/user-attachments/files/19898306/1745524555403-player-script.zip)

**Comment by JustNOTRO** [comment]
[1745564049775-player-script.zip](https://github.com/user-attachments/files/19904126/1745564049775-player-script.zip)

[1745564049783-player-script.zip](https://github.com/user-attachments/files/19904130/1745564049783-player-script.zip)

I'm receiving this logs when trying to search & play a youtube video.
This is my code:

`async function playSong(interaction, voiceChannel) {
    const query = interaction.options.getString('query');

    try {
        await interaction.reply('Playing song!!');

        const distube = new DisTube(interaction.client, {
            plugins: [new YouTubePlugin(), new YtDlpPlugin()],
            emitNewSongOnly: true,
        });

        await distube.play(voiceChannel, query);
    } catch (err) {
        console.error('Playback error:', err);
        await interaction.error('Could not play the requested song.');
    }
    
}`

**Comment by electricduckpond** [comment]
Try use the fix provided in this issue https://github.com/distubejs/ytdl-core/issues/245 It worked for me. you just have to replace the sig.js file

**Comment by JustNOTRO** [comment]
Hey, @electricduckpond, Thanks for replying, I've tried the changes you suggested, the same issue persists.

**Comment by Rebot281208** [comment]
[1745615974785-player-script.zip](https://github.com/user-attachments/files/19916666/1745615974785-player-script.zip)

**Comment by akashdas2004** [comment]
WARNING: Could not parse decipher function.
Stream URLs will be missing.
Please report this issue by uploading the "1745620609047-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
WARNING: Could not parse n transform function.
Please report this issue by uploading the "1745620609054-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

**Comment by HolyZephyros** [comment]
WARNING: Could not parse decipher function.
Stream URLs will be missing.
Please report this issue by uploading the "1745863657791-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.

[1745863657791-player-script.zip](https://github.com/user-attachments/files/19946245/1745863657791-player-script.zip)

**Comment by BobbyHertz** [comment]
Happening again as of today, 4.16.9.

**Comment by Method-Development** [comment]
Happening again, even with the signature replacement method from before

**Comment by rwul91** [comment]
1745930547760-base.js

**Comment by BobbyHertz** [comment]
Current fix as of this time is the gist posted here:

https://github.com/distubejs/ytdl-core/issues/249

**Comment by joe-arms** [comment]
[1745983707660-player-script.zip](https://github.com/user-attachments/files/19970652/1745983707660-player-script.zip)

**Comment by Cedrugs** [comment]
[Archive.zip](https://github.com/user-attachments/files/19974969/Archive.zip)

**Comment by ElliotJS** [comment]
[1746175407074-player-script.zip](https://github.com/user-attachments/files/20010485/1746175407074-player-script.zip)

**Comment by opalnext** [comment]
[1746196212431-player-script.zip](https://github.com/user-attachments/files/20014186/1746196212431-player-script.zip)

**Comment by M3canixx** [comment]
[1746214553184-player-script.zip](https://github.com/user-attachments/files/20017606/1746214553184-player-script.zip)

**Comment by TorresOwO** [comment]
C:\Users\Sergio\Desktop\node_projects\spotip17an_back\1746383266446-player-script.js

**Comment by TorresOwO** [comment]
[1746383266446-player-script.zip](https://github.com/user-attachments/files/20029044/1746383266446-player-script.zip)

**Comment by RuzitoTV** [comment]
[1746383670718-player-script.zip](https://github.com/user-attachments/files/20029050/1746383670718-player-script.zip)

**Comment by guilebc** [comment]
[1746386976627-player-script.zip](https://github.com/user-attachments/files/20029553/1746386976627-player-script.zip)

**Comment by MadProbe** [comment]
[1746388807057-player-script.js.zip](https://github.com/user-attachments/files/20029660/1746388807057-player-script.js.zip)

**Comment by xvaie** [comment]
[1746457825230-player-script.js.zip](https://github.com/user-attachments/files/20039192/1746457825230-player-script.js.zip)

**Comment by 1Classy** [comment]
[1746511829167-player-script.zip](https://github.com/user-attachments/files/20053449/1746511829167-player-script.zip)

**Comment by LAMAgalletta0IQ** [comment]
[1746538793027-player-script.zip](https://github.com/user-attachments/files/20063155/1746538793027-player-script.zip)

**Comment by FabainHioc** [comment]
1746575333019-player-script.js

**Comment by MikeTodd** [comment]
Trying to play this song: https://www.youtube.com/watch?v=DiYTaQ-Mgck

Got the following:

```
2025-05-07 16:22:26 WARNING: Could not parse decipher function.
2025-05-07 16:22:26 Stream URLs will be missing.
2025-05-07 16:22:26 Please report this issue with the "1746649346213-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
2025-05-07 16:22:26 WARNING: Could not parse n transform function.
2025-05-07 16:22:26 Please report this issue with the "1746649346223-base.js" file on https://github.com/distubejs/ytdl-core/issues/144.
```

So, reporting and uploading both files (in contained ZIP) as directed. :) 

[1746649346223-base.zip](https://github.com/user-attachments/files/20093022/1746649346223-base.zip)

**Comment by DMCK96** [comment]
[1746871911722-base.zip](https://github.com/user-attachments/files/20138946/1746871911722-base.zip)

**Comment by Godbot** [comment]
[1748062987299-player-script.zip](https://github.com/user-attachments/files/20421338/1748062987299-player-script.zip)

**Comment by Yamac-Koc** [comment]
[1749520266638-player-script.txt](https://github.com/user-attachments/files/20663537/1749520266638-player-script.txt)

PLEASE FIX ASAP THIS IS ANNOYING!

**Comment by nphcm2003** [comment]
[1749536642544-player-script.txt](https://github.com/user-attachments/files/20666202/1749536642544-player-script.txt)

![Image](https://github.com/user-attachments/assets/2370e7ce-4f2b-4cff-8a91-2098165c3fd7)

i have updated my ytdl-core to latest, is this a problem with my ffmpeg, i install it via cmd with "winget install ffmpeg"

**Comment by PixelZiggy** [comment]
[1749593010406-player-script.zip](https://github.com/user-attachments/files/20680588/1749593010406-player-script.zip)

**Comment by Linkiddrak2** [comment]
[1749590194737-player-script.zip](https://github.com/user-attachments/files/20682138/1749590194737-player-script.zip)

**Comment by armandinhox** [comment]
[1749604912181-player-script.zip](https://github.com/user-attachments/files/20682582/1749604912181-player-script.zip)

**Comment by ghfhffh12345** [comment]
This has already been fixed at [@nuclearplayer/ytdl-core](https://github.com/NuclearPlayer/ytdl-core). Hope this helps.

**Comment by KaedeYure** [comment]
> This has already been fixed at [@nuclearplayer/ytdl-core](https://github.com/NuclearPlayer/ytdl-core). Hope this helps.

You are a lifesaver, thanks!!

**Comment by zachisfine** [comment]
[1749644530057-player-script.zip](https://github.com/user-attachments/files/20690835/1749644530057-player-script.zip)

**Comment by zachisfine** [comment]
It seems to be only on some youtube links, e.g.:

https://www.youtube.com/watch?v=fpv1tUHO2Es does not work

but

https://www.youtube.com/watch?v=E8c83bpOVXo works

**Comment by pxldevv** [comment]
> > This has already been fixed at [@nuclearplayer/ytdl-core](https://github.com/NuclearPlayer/ytdl-core). Hope this helps.
> 
> wait, is the error fixed? How do i implement it in my project?? Please help.

Someone else forked this project and fixed it themselves. They made it public so that other people can also use this fix, me personally I am waiting on distube to update it in theirs.

**Comment by pxldevv** [comment]
Oh didn't realize but it looks like they updated it today. @Yamac-Koc (You can do npm update in your root folder)

**Comment by calini** [comment]
Hi, having issues with this using [museofficial/muse](https://github.com/museofficial/muse), which uses [distubejs/ytdl-core](https://github.com/distubejs/ytdl-core):

[1749914771246-player-script.zip](https://github.com/user-attachments/files/20738482/1749914771246-player-script.zip)

Any chance you cool look into it? Cheers!

**Comment by Quema100** [comment]
> Hi, having issues with this using [museofficial/muse](https://github.com/museofficial/muse), which uses [distubejs/ytdl-core](https://github.com/distubejs/ytdl-core):
> 
> [1749914771246-player-script.zip](https://github.com/user-attachments/files/20738482/1749914771246-player-script.zip)
> 
> Any chance you cool look into it? Cheers!

update the library to the latest version.

Current version:4.16.12

Updating should resolve the issue.

**Comment by lotus64yt** [comment]
[1750659870840-base.js.txt](https://github.com/user-attachments/files/20859542/1750659870840-base.js.txt)

**Comment by Mace12409** [comment]
I dont know at all what am doing but am js follow docs and logs here is my file below.
`could not upload it in git here is the link to the download`

https://files.macehub.club/view/v8GIUU.js

**Comment by techlvling** [comment]
muse-bot | WARNING: @distube/ytdl-core is out of date! Update with "npm install @distube/ytdl-core@latest".
muse-bot | WARNING: Could not parse decipher function.
muse-bot | Stream URLs will be missing.
muse-bot | Please report this issue by uploading the "1754515788058-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
muse-bot | WARNING: Could not parse n transform function.
muse-bot | Please report this issue by uploading the "1754515788088-player-script.js" file on https://github.com/distubejs/ytdl-core/issues/144.
